tests/tstackarrays.nim

```diff
@@ -34,3 +34,27 @@ suite "Stack arrays":
     expect RangeError:
       arr[3] = "another test"
 
+  test "proof of stack allocation":
+    proc fun() =
```

Comment:
This test doesn't check for the stack allocation. It checks that `arr` variable (which holds an address of the allocated array (be it on stack or on the heap)) is between `x1` and `x2` variables. I'd suggest to delete this test in favor of nothing.

Reply:
Ok, one possible way to test this is check that addr of an element is no further than (let's say) 1KB from a local stack variable.
```nim
assert(abs(addr localDummy - addr myStackArray[1]) < 1024)
```

Reply:
[feel free to ignore...]
maybe with this it would become 100% correct? (untested):
replace 
`let p_addr = cast[int](addr arr)`
by
`let p_addr = cast[int](addr arr[1])`

it should work since x1, x2 are on the stack.

> assert(abs(addr localDummy - addr myStackArray[1]) < 1024)

this one makes un-necessary assumptions